BootloaderCorePkg/Library/BootloaderLib/BootloaderLib.c

```diff
@@ -53,48 +92,30 @@ GetComponentHash (
     return RETURN_NOT_FOUND;
   }
 
-  *HashData = (UINT8 *)HashStorePtr->Data[HashIndex].Data;
-
-  return RETURN_SUCCESS;
-}
-
-/**
-  Set the component hash data by the component type.
-
-  @param[in]  ComponentType   Component type.
-  @param[in] HashData        Hash data pointer corresponding Component type.
-
-  @retval RETURN_SUCCESS             Set hash data succeeded.
-  @retval RETURN_UNSUPPORTED         Hash component type is not supported.
-  @retval RETURN_NOT_FOUND           Hash data is not found.
-  @retval RETURN_INVALID_PARAMETER   HashData is NULL.
+  HashEntryPtr = HashStorePtr->Data;
 
-**/
-RETURN_STATUS
-SetComponentHash (
-  IN       UINT8             ComponentType,
-  IN CONST UINT8            *HashData
-  )
-{
-  LOADER_GLOBAL_DATA  *LdrGlobal;
-  HASH_STORE_TABLE    *HashStorePtr;
-  UINT8               HashIndex;
+  for (Index = 0; Index < HashStorePtr->DigestCount; Index++){
 
-  if (HashData == NULL) {
-    return RETURN_INVALID_PARAMETER;
+    HashEntryData = (HASH_STORE_DATA *) HashEntryPtr;
+    if(HashEntryData->Usage & (1 << HashIndex)){
+      DEBUG((DEBUG_INFO, "HASH Entry found 0x%x\n",Index));
+      //Hash Entry found
+      break;
+    } else {
+      HashEntryPtr +=  sizeof(HASH_STORE_DATA) + HashEntryData->DigestLen;
+    }
   }
 
-  HashIndex = ComponentType;
-  LdrGlobal = GetLoaderGlobalDataPointer ();
-  HashStorePtr = (HASH_STORE_TABLE *)LdrGlobal->HashStorePtr;
-  if ((HashStorePtr == NULL) || (HashIndex != COMP_TYPE_PAYLOAD_DYNAMIC)) {
-    return RETURN_UNSUPPORTED;
+  if (Index == HashStorePtr->DigestCount){
+    return RETURN_NOT_FOUND;
+  } else {
+    *HashData = (UINT8 *)HashEntryData->Digest;

```

Comment:
Check if HashData/HashAlg is NULL pointer before assignment.

Reply:
will do.

Reply:
where did you check  HashAlg is NULL or not ?

Reply:
Done